WordPress/Classes/ViewRelated/Blog/BlogDetailHeaderView.m

```diff
@@ -0,0 +1,116 @@
+const CGFloat BlogDetailHeaderViewLabelHeight = 20.0;
+const CGFloat BlogDetailHeaderViewLabelHorizontalPadding = 10.0;
+
+@interface BlogDetailHeaderView ()
+
+@property (nonatomic, strong) UIImageView *blavatarImageView;
+@property (nonatomic, strong) UILabel *titleLabel;
+@property (nonatomic, strong) UILabel *subtitleLabel;
+
+@end
+
+@implementation BlogDetailHeaderView
+
+- (instancetype)initWithFrame:(CGRect)frame
+{
+    self = [super initWithFrame:frame];
+    if (self) {
+        _blavatarImageView = [self imageViewForBlavatar];
+        [self addSubview:_blavatarImageView];
+
+        _titleLabel = [self labelForTitle];
+        [self addSubview:_titleLabel];
+
+        _subtitleLabel = [self labelForSubtitle];
+        [self addSubview:_subtitleLabel];
+
+        [self configureConstraints];
+    }
+    return self;
+}
+
+#pragma mark - Public Methods
+
+- (void)setBlog:(Blog *)blog
+{
+    [self.blavatarImageView setImageWithBlavatarUrl:blog.blavatarUrl isWPcom:blog.isWPcom];
+
+    // if the blog name is missing, we want to show the blog displayURL instead
+    [self.titleLabel setText:(blog.blogName ? blog.blogName : blog.displayURL)];
+    [self.subtitleLabel setText:blog.displayURL];
+}
+
+#pragma mark - Private Methods
+
+- (void)configureConstraints
+{
+    NSDictionary *views = NSDictionaryOfVariableBindings(_blavatarImageView, _titleLabel, _subtitleLabel);
+    NSDictionary *metrics = @{@"blavatarSize": @(BlogDetailHeaderViewBlavatarSize),
+                              @"labelHeight":@(BlogDetailHeaderViewLabelHeight),
+                              @"labelHorizontalPadding": @(BlogDetailHeaderViewLabelHorizontalPadding)};
+    [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[_blavatarImageView(blavatarSize)]|"
+                                                                 options:0
+                                                                 metrics:metrics
+                                                                   views:views]];
+    [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"|[_blavatarImageView(blavatarSize)]-labelHorizontalPadding-[_titleLabel]|"
+                                                                 options:0
+                                                                 metrics:metrics
+                                                                   views:views]];
+    [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"|[_blavatarImageView(blavatarSize)]-labelHorizontalPadding-[_subtitleLabel]|"
+                                                                 options:0
+                                                                 metrics:metrics
+                                                                   views:views]];
+    [self addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:|[_titleLabel(labelHeight)]-0-[_subtitleLabel(labelHeight)]"
```

Comment:
No need to define a 0 spacer unless you need to set its priority. You could just omit `-0-` and it would mean the same.